contracts/NonfungibleTokenPositionDescriptor.sol

```diff
@@ -48,4 +64,24 @@ contract NonfungibleTokenPositionDescriptor is INonfungibleTokenPositionDescript
                 })
             );
     }
+
+    function hasToken0RatioNumerator(address token0, address token1) public view returns (bool) {
+        if (ratioNumeratorTokens[token1]) {
+            return false;
+        } else if (ratioNumeratorTokens[token0]) {
+            return true;
+        } else if (ratioDenominatorTokens[token1]) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    function addRatioNumeratorToken(address token) external onlyOwner() {
```

Comment:
aww crap....just realizing that if we use the zeppelin `TransparentProxy`  governance cannot both be the admin for upgradeability and _also_ call to the implementation contract... we may have to overwrite that if we want governance to be able to assign more of these..